mypy/errors.py

```diff
@@ -351,7 +351,9 @@ def add_error_info(self, info: ErrorInfo) -> None:
         self._add_error_info(file, info)
 
     def is_ignored_error(self, line: int, info: ErrorInfo, ignores: Dict[int, List[str]]) -> bool:
-        if line not in ignores:
+        if info.code and info.code.enabled is False:
+            return True
```

Comment:
You can pass the enabled and disabled error codes to `Errors`, and we can use them together with the `default_enabled` attribute to decide whether to ignore them or not.

This way there will be less global mutable state, and it will be easier to have per-file enabled/disabled sets in the future.